build-tools/automation/azure-pipelines.yaml

```diff
@@ -218,6 +218,14 @@ stages:
           /bl:$(System.DefaultWorkingDirectory)\bin\Test$(XA.Build.Configuration)\msbuild-run-ji-tests.binlog
       condition: succeededOrFailed()
 
+    - powershell: |
+        Stop-Process -Name nunit3-console
+        Stop-Process -Name xabuild
+        Stop-Process -Name MSBuild
+        Stop-Process -Name adb
+      displayName: kill leftover processes
+      condition: succeededOrFailed()
```

Comment:
I think we need to change both this and the condition below to `always()` or `eq(variables['Agent.JobStatus'], 'Canceled')`. When a timeout occurs, the job will be cancelled, and `succeededOrFailed()` will evaluate as false.